### PR TITLE
Review widget refactor to stateful

### PR DESCRIPTION
**Pull request by NylonEric**
* GET reviews route working with postman
* Working route for ReviewsMeta API call from postman
* Removed GET reviews request parameters to fix bug in initial API bundle
* eslint fixes
* fixed product_id query variable bug in server.js
* productId state passing and updating RatingsAndReviews component
* more reviews button implemented
* reviews meta get request implemented and passing to props
* fix NaN bug when no reviews meta available
* added API calls for review reporting and helpful

### CODE REVIEW

File: client/__tests__/RatingsAndReviews.test.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ var onChangeFileHandler = (event) => {
     selectedImageFile: event.target.files[0],
     fileLoaded: 0
   });
-}
+};
```
Comment (milofultz) [inline client/__tests__/RatingsAndReviews.test.js:12]: 👀

File: client/src/components/ProductDetailPage.jsx [hunk 2]
```diff
@@ -59,6 +62,14 @@ class ProductDetailPage extends React.Component {
       });
   }
 
+  getCurentReviewsMetaFromAPI(productId = this.state.currentProductId) {
+    console.log('getCurrent Meta');
```
Comment (milofultz) [inline client/src/components/ProductDetailPage.jsx:66]: remove logs? Not sure if you want it here

File: client/src/components/ProductDetailPage.jsx [hunk 3]
```diff
@@ -90,6 +101,7 @@ class ProductDetailPage extends React.Component {
     this.getCurrentProductFromAPI(productId)
       .then(productInformation => {
         updatedProducts[productId] = productInformation;
+        console.log('load product information', productInformation);
```
Comment (milofultz) [inline client/src/components/ProductDetailPage.jsx:104]: same

File: client/src/components/RatingsAndReviews/RatingsAndReviews.jsx [hunk 4]
```diff
@@ -1,75 +1,173 @@
...
+
+  // reviewHelpful(review_id) {
+
+
+  // }
+
+  // sorting and filter handlers:
+  // populate state with reviews:
+  setReviews(reviews) {
+    console.log('setReviews:', reviews);
```
Comment (milofultz) [inline client/src/components/RatingsAndReviews/RatingsAndReviews.jsx:66]: same

File: client/src/components/RatingsAndReviews/RatingsAndReviews.jsx [hunk 5]
```diff
@@ -1,75 +1,173 @@
...
+    this.setState({
+      _reviews: reviews,
+      reviewsLength: reviews.length,
+      reviewsToShow: newReviewsToShow,
+      reviewsToShowLength: 2
+    });
+  }
+
+  setReviewsLength() {
+    let currentLength = this.state.reviews.length;
```
Comment (milofultz) [inline client/src/components/RatingsAndReviews/RatingsAndReviews.jsx:77]: I don't see a `reviews` in state, only a `_reviews`

File: client/src/components/RatingsAndReviews/RatingsAndReviews.jsx [hunk 6]
```diff
@@ -1,75 +1,173 @@
...
+    this.setState({
+      reviewsLength: currentLength
+    });
+  }
+
+  setReviewsToShowLength() {
+    console.log('setReviewsToShowLength', this.state.reviewsToShowLength);
```
Comment (milofultz) [inline client/src/components/RatingsAndReviews/RatingsAndReviews.jsx:84]: same

File: client/src/components/RatingsAndReviews/RatingsAndReviews.jsx [hunk 7]
```diff
@@ -1,75 +1,173 @@
...
+      currentLength === this.state.reviewsLength - 1 ? currentLength += 1 : currentLength += 2;
+      this.setState({
+        reviewsToShowLength: currentLength
+      });
+    }
+  }
+
+  moreReviewsButtonHandler(e) {
+    // e.preventDefault();
+    console.log('moreReviewsButtonHandler');
```
Comment (milofultz) [inline client/src/components/RatingsAndReviews/RatingsAndReviews.jsx:97]: same, can remove event from handler also, if it's not used

File: client/src/components/RatingsAndReviews/RatingsAndReviews.jsx [hunk 8]
```diff
@@ -1,75 +1,173 @@
...
+      reviews,
+      reviewsMeta,
+    } = this.props;
+
+    let rating = calculateRating(reviewsMeta.ratings);
+    let moreReviews = this.state.reviewsLength >= this.state.reviewsToShowLength;
+
+    return (
+      <div id="ratings-main" data-testid="ratings-main">
+        {/* {console.log('rendering Ratings&Reviews component')} */}
```
Comment (milofultz) [inline client/src/components/RatingsAndReviews/RatingsAndReviews.jsx:115]: remove

File: client/src/components/RatingsAndReviews/RatingsAndReviews.jsx [hunk 9]
```diff
@@ -1,75 +1,173 @@
...
-              top.showModal(
-              <AddReviewModal
-                top={top}
-                productId={productId}
-                currentProduct={currentProduct}
-              />);
-            }}>ADD A REVIEW +</button>
+          <ReviewsList
+            reviews={this.state._reviews.length >= 2 ? this.state._reviews.slice(0, this.state.reviewsToShowLength) : reviews}
+            // reviewHelpful={reviewHelpful}
```
Comment (milofultz) [inline client/src/components/RatingsAndReviews/RatingsAndReviews.jsx:144]: remove these probably

File: client/src/components/RatingsAndReviews/ReviewsList.jsx [hunk 10]
```diff
@@ -1,11 +1,24 @@
 import React from 'react';
 import Review from './Review.jsx';
+import axios from 'axios';
 
 const ReviewsList = (props) => {
+  const reviewHelpful = (reviewId) => {
+    axios.put('/reviews/:review_id/helpful', { review_id: reviewId })
+      .then(info => console.log('info:', info))
```
Comment (milofultz) [inline client/src/components/RatingsAndReviews/ReviewsList.jsx:8]: this console log might throw?

File: client/src/components/RatingsAndReviews/ReviewsList.jsx [hunk 11]
```diff
@@ -1,11 +1,24 @@
...
+      .catch(err => err);
+  };
+
+  const reportReview = (reviewId) => {
+    axios.put('/reviews/:review_id/report', { review_id: reviewId })
+      .then(info => console.log('info:', info))
```
Comment (milofultz) [inline client/src/components/RatingsAndReviews/ReviewsList.jsx:14]: same here

File: server/atelier-queries.js [hunk 12]
```diff
@@ -195,6 +201,7 @@ const getCurrentProductBundle = (productId, headers) => {
 };
 
 const getRelatedProductsBundle = (relatedProducts, headers) => {
+  console.log('relatedProducts in bundle:', relatedProducts[0]);
```
Comment (milofultz) [inline server/atelier-queries.js:204]: log

File: server/server.js [hunk 13]
```diff
@@ -193,8 +192,8 @@ app.get('/products/related', (req, res) => {
 // Specific and smaller queries
 // .../reviews?id=12345
 app.get('/reviews', (req, res) => {
-  const productId = req.query.id;
-  atelierQueries.getProductReviews(productId, atelierHeaders)
+  const query = req.query.product_id;
```
Comment (milofultz) [inline server/server.js:195]: nit, but I'm seeing `productId` and `product_id`, not sure if they can/should be reconciled, but something to consider

File: server/server.js [hunk 14]
```diff
@@ -253,21 +252,22 @@ app.post('/reviews/:product_id', uploadS3.array('images', 5), (req, res) => {
...
+      .then(result => res.end(JSON.stringify(result)))
+      .catch(error => {
+        console.error(error);
+        res.end(JSON.stringify(error));
+      });
   }
 });
 
 app.put('/reviews/:review_id/helpful', (req, res) => {
+  console.log('mark review as helpful - server');
```
Comment (milofultz) [inline server/server.js:270]: remove logs

File: server/server.js [hunk 15]
```diff
@@ -279,10 +279,10 @@ app.put('/reviews/:review_id/helpful', (req, res) => {
 });
 
 app.put('/reviews/:review_id/report', (req, res) => {
+  console.log('report review - server');
```
Comment (milofultz) [inline server/server.js:282]: logs

**Review by milofultz — CHANGES_REQUESTED** [review]
almost everything is a nit, but just the one place where I think it might error out on `_reviews` when state has `reviews`.

File: client/src/components/RatingsAndReviews/RatingsAndReviews.jsx (hunk 5 above)
Comment (NylonEric) [reply client/src/components/RatingsAndReviews/RatingsAndReviews.jsx:77]: Good catch.

File: client/src/components/RatingsAndReviews/ReviewsList.jsx (hunk 11 above)
Comment (NylonEric) [reply client/src/components/RatingsAndReviews/ReviewsList.jsx:14]: I'm trying to maintain consistency with the API endpoints to avoid confusion.

**Comment by NylonEric** [comment]
All changes made except API key name as noted.

**Review by milofultz — APPROVED** [review]

**Pull request merged**